go.mod

```diff
@@ -0,0 +1,25 @@
+module github.com/uber/tchannel-go
+
+go 1.16
+
+require (
+	github.com/HdrHistogram/hdrhistogram-go v1.1.2 // indirect
+	github.com/apache/thrift v0.0.0-20161221203622-b2a4d4ae21c7
```

Comment:
Can you use the 0.9.3 version here? (Not sure if the apache tag works with go mod, I remember having issues in the past,
https://issues.apache.org/jira/browse/THRIFT-5205
Didn't end up following up though).

Either way, I'd suggest adding a comment with the specific constraint that previously was in glide.yaml and add comments calling out that it won't work with newer versions (and why not, etc).

Reply:
pinned to the same sha as glide.lock, was there some other reason why we can't be on a newer version?

Reply:
https://issues.apache.org/jira/browse/THRIFT-5205 is why we can't use a v0.9 in the version for the dep

the reason we don't support newer versions is that Thrift changed the interface to add context to a bunch of interfaces, which then requires breaking changes to tchannel-go. breaking change + major version change resulted in diamond dependency problems (service A wants v1 of tchannel, but a library it depends on wants v2). There wasn't a feasible upgrade path before, but maybe it's more feasible now (if there's only a few repos depending on tchannel)

Reply:
ah, gotcha. thanks for the explanation! I'll look into the upgrade issue separately.